.azure-pipelines/client.yml

```diff
@@ -70,3 +70,7 @@ jobs:
         publishJUnitResults: true
         testResultsFiles: '**/TEST-*.xml'
         goals: 'test'
+
+    - task: ms.vss-governance-buildtask.governance-build-task-component-detection.ComponentGovernanceComponentDetection@0
```

Comment:
Please move to a separate parallel job like https://github.com/Azure/azure-sdk-for-js/blob/master/.azure-pipelines/client.yml#L55-L84

Reply:
While you have moved this to a separate job currently it doesn't do anything because it is skipped in PR validation but when it is run there aren't any projects built so you still need to run the build. 

@mikeharder do you think it is worth having this be a separate Job? It seems like this would be good to be at the end of one of the test jobs so we don't end up doing too much duplicate work.

Reply:
@weshaggard I like the concept of a separate static analysis/audit job. If we make it a dependent of `Build` and make the tests take a dependency on THAT, we will save resources on build or audit failure.

Reply:
I like the general idea of keeping them separate but given there isn't a great way to share produced assets between jobs you will still have to re-run the build in the Audit leg to do the analysis. Not a huge deal right now but something to consider, we do the same thing for testing legs also.

Reply:
I'm not an expert on CG, I just gave the docs a read through this morning. From what I can tell aren't the generated artifacts present after that job?  Your concern is with the generated _artifacts_ not being component governed yes?

Reply:
Yes that is correct. If we can convince ourselves that the task is just using whatever is in the build artifacts including the ones produced by the earlier jobs then this approach may work.

Reply:
It should be easy to test. Just try to understand where the task gets the stuff it analyzes.

Reply:
My preference is for our pipelines to have three general "phases":

1. `build:` Build the final package as quickly as possible, without running tests or static analysis.
2. `static analysis`: Run static analysis tools like dependency checker, linter/FxCop/StyleCop, or ComponentGovernance.  May run in parallel with `build` for faster results, or may depend on `build` to avoid wasting build agent time if `build` fails.
3. `test`: Run tests on many OS and platform versions.  Should depend on `build` to avoid wasting build agent time if `build` fails, especially since `test` can use 10 or more agents.

Reply:
I like the general phases and I think I will add a version of this to the guideline document. I think I would make it a little stronger for (2) and (3) and say it should always depend on build/package and if possible should reuse the assets produced in the build/package job to avoid duplicated work.

Reply:
Thank you @weshaggard  and @mikeharder . Should we also have the test phase depend on the 'Audit' phase?

Reply:
I don't think we need to have the test job depend on the audit job we can run those in parallel.

Reply:
No, I would just have `test` depend on `build`.  Even if static analysis / audit fails, we still want to get test results.  However if build fails there's little reason to try running static analysis or tests, since the results will basically be invalid.